Warning cleanup

rillian opened the pull request:
Resolve the majority of errors reported by pycodestyle/pytest/setup.py. Tested under python 3.7.

Please consider merging these. I was trying to figure out why HookTest wasn't reporting anything on my test corpus and then started looking at the code, but I wanted to be able to see any new errors from changes I made debugging. Reducing the warning output makes that easier.

This PR reduces `pycodestyle --ignore E501 *.py HookTest` from 48 lines to 8.

PonteIneptique commented:
BTW, if this continues to not report anything, feel free to share the corpus in an issue and I'll gladly see if HookTest is missing something or is bugged :)

rillian commented on tests/test_units.py:
```diff
@@ -343,7 +343,7 @@ def setUp(self):
         self.Epidoc = HookTest.capitains_units.cts.CTSText_TestUnit("/false/path")
         self.Epidoc.scheme = "epidoc"
-        self.frame = """<TEI xmlns="http://www.tei-c.org/ns/1.0">
+        self.frame = r"""<TEI xmlns="http://www.tei-c.org/ns/1.0">
```
That's exactly why a raw string is appropriate: The regular expressions in the xml should be passed as part of an xml document, not expanded as a python string first. The data passes through ok, because `\w` isn't a Python string escape, but since Python 3.6 the interpreter warns about it in case they want to add such an escape in the future.

If you had:

    "<p>Use the <code>\n</code> newline escape to insert blank lines.</p>"

Letting the python interpreter replace the `\n` with an actual newline would result in incorrect rendering. It's the same with the xpath regex, unless I'm missing something?

rillian replied:
Anything else I can do to address your concern here, @PonteIneptique?

PonteIneptique replied:
I'll need to review the issue you talked about but won't have the time before tomorrow (EU Time) I believe

rillian replied:
Ok, thanks!

PonteIneptique replied:
I am not able to reproduce the warning on the `\w` you are talking about unfortunately...

PonteIneptique replied:
I read https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
So even if there is no warning, I'll have to follow you on this one. Even though I have to say I am not agreeing with the path taken by python on this one ;)

rillian commented:
Thanks for the quick review!

> As long as you add these clean-ups, can you automate these checkings in the .travis.yml ? Thanks in advance ;)

I've gone ahead and fixed the remaining style warnings (except for long lines) and added pycodestyle to the travis run. It will new report failures for any new warnings that are introduced.

rillian commented:
I think the coverage changes might be rounding error? I can't find a line where the coverage is different.

PonteIneptique commented:
I am evaluating this as I write. Sorry for this huge delay.

PonteIneptique commented:
Thanks for this Pull Request and sorry for the delay ;)

rillian commented:
Thanks for reviewing. It was an extensive set of changes. To me it sometimes seems like Python strings are as confusing as Rust strings!

PonteIneptique commented:
What is the most confusing to me with this introduction is how this will affect stuff such as f.read() containing such characters. Obviously, it will (or maybe not) not throw errors, but then, two strings have different behavior...

rillian commented:
I think the invalid escape sequence warnings (soon to be errors) only apply to string *literals*. That is, strings created from text in surrounded by quotes in a Python file. That's where the backslash-escape system is applied to create strings with characters which aren't easy to represent.

Strings created through f.read() come off the disk already containing whatever such characters they do, and escape sequences aren't interpreted. They can still raise an exception and fail to create a string if they're [not valid in the expected character encoding](http://kunststube.net/encoding/), though.
